tests/testsupport/broker.py

```diff
@@ -0,0 +1,62 @@
+import socket
+
+import pytest
+
+
+class FakeBroker:
+    def __init__(self):
+        # Bind to "localhost" for maximum performance, as described in:
+        # http://docs.python.org/howto/sockets.html#ipc
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        sock.settimeout(30)
+        sock.bind(("localhost", 1888))
+        sock.listen(1)
+
+        self._sock = sock
+        self._conn = None
+
+    def start(self):
+        if self._sock is None:
+            raise ValueError('Socket is not open')
+
+        (conn, address) = self._sock.accept()
+        conn.settimeout(10)
+        self._conn = conn
+
+    def finish(self):
+        if self._conn is not None:
+            self._conn.close()
+            self._conn = None
+
+        if self._sock is not None:
+            self._sock.close()
+            self._sock = None
+
+    def expect_packet(self, packet_expected):
+        if self._conn is None:
+            raise ValueError('Connection is not open')
+
+        packet_in = self._conn.recv(len(packet_expected))
+        if not packet_in:
+            raise RuntimeError('Connection was closed')
+        assert packet_in == packet_expected
```

Comment:
Currently if the assert fail, the message looks like:
```
self = <testsupport.broker.FakeBroker instance at 0x7f655b701d88>, packet_expected = '\x10#\x00\x06MQIsdp\x03\x02\x00<\x00\x1502-con-discon-success'

    def expect_packet(self, packet_expected):
[...]
>       assert packet_in == packet_expected
E       AssertionError
```
e.g. we don't have at all the packet_in value and expected packet is show un-decoded.

We should at least show the packet_in value. I think of two way to do that:
* always print() packet received/sent/expected. Since py.test capture stdout and only show it on error it should not generate noise and may help to understand failure since all network exchange will be logged.
* Add a assert message. Something like:
```
assert packet_in == packet_expected, "Expected = %s\nReceived = %s" % (decoded_expected, decoded_received)
```